frontend/src/features/comparisons/Comparison.tsx

```diff
@@ -137,13 +148,34 @@ const Comparison = ({
         .then((comparison) => {
           setInitialComparison(comparison);
           setIsLoading(false);
+          initializing.current = false;
         })
         .catch(() => {
           setInitialComparison(null);
           setIsLoading(false);
+          initializing.current = false;
         });
   }, [pollName, uidA, uidB, selectorA.uid, selectorB.uid]);
 
+  /**
+   * When the UI language changes, refresh the ratings to retrieve the
+   * corresponding localized entity contexts.
+   */
+  useEffect(() => {
+    if (initializing.current || isLoading) {
```

Comment:
I don't understand why this check on `initliazing.current` is useful here :thinking:

Reply:
I checked and it's not useful, the `isLoading` check seems to be enough.